### PR TITLE
context: parse with high level function

### PR DESCRIPTION
**Pull request by jeanseb6wind**
Context parsing should use high level function instead of low level.

### CODE REVIEW

File: libyang/context.py [hunk 1]
```diff
@@ -340,42 +340,27 @@ def parse_data(
...
 
-        if parent is not None:
-            ret = lib.lyd_parse_data(
-                self.cdata,
-                parent.cdata,
-                data[0],
-                fmt,
-                parser_flgs,
-                validation_flgs,
-                ffi.NULL,
```
Comment (rjarry) [inline libyang/context.py:360]: If I'm not mistaken, `parent` is now ignored.

File: tests/test_data.py [hunk 2]
```diff
@@ -122,30 +122,6 @@ def test_data_add_path(self):
...
-
-        dnode = self.ctx.parse_data(
-            "json",
-            in_data=self.JSON_CONFIG_FILE,
-            in_type=IOType.FILEPATH,
-            validation_no_state=True,
-        )
-        self.assertIsInstance(dnode, DContainer)
-        dnode.free()
-
```
Comment (rjarry) [inline tests/test_data.py:148]: Why did you remove this test?

**Review by rjarry — CHANGES_REQUESTED** [review]
You did not explain why you made this change. Could you motivate a bit?

Thanks

**Comment by jeanseb6wind** [comment]
It's ok now.
We have to use the libyang wrapper around `parse_data` insead of using `parse_data` directly

**Comment by rjarry** [comment]
You only explained *what* your commit does. Not *why* you made it. I don't understand why this is needed.

**Comment by jeanseb6wind** [comment]
Hello @rjarry, is it ok for you now that I have updated the commit message ?

File: libyang/context.py [hunk 3]
```diff
@@ -424,7 +440,33 @@ def parse_data_file(
...
+        parser_opaq: bool = False,
+        parser_ordered: bool = False,
+        parser_strict: bool = False,
+        validation_no_state: bool = False,
+        validation_validate_present: bool = False,
+    ) -> Optional[DNode]:
+        return self.parse_data(
+            fmt,
+            in_type=IOType.FILEPATH,
+            in_data=fileobj,
```
Comment (rjarry) [inline libyang/context.py:469]: This should call `lyd_parse_data_filepath` directly. I don't see the point in wrapping the call into a generic parse_data that ends up calling specific functions.

File: pylintrc [hunk 4]
```diff
@@ -88,6 +88,7 @@ disable=
 	too-many-statements,
 	unused-argument,
 	wrong-import-position,
+	unnecessary-lambda,
```
Comment (rjarry) [inline pylintrc:91]: I think this is not necessary anymore.

File: libyang/context.py [hunk 5]
```diff
@@ -424,7 +440,33 @@ def parse_data_file(
...
-            parent=parent,
+            parser_lyb_mod_update=parser_lyb_mod_update,
+            parser_no_state=parser_no_state,
+            parser_parse_only=parser_parse_only,
+            parser_opaq=parser_opaq,
+            parser_ordered=parser_ordered,
+            parser_strict=parser_strict,
+            validation_no_state=validation_no_state,
+            validation_validate_present=validation_validate_present,
+        )
```
Comment (rjarry) [inline libyang/context.py:451]: This should call `lyd_parse_data_fd` directly. I don't see the point in wrapping the call into a generic parse_data that ends up calling specific functions.

**Review by rjarry — CHANGES_REQUESTED** [review]
I finally understand the idea behind this patch :) Please see my comments.

File: libyang/context.py [hunk 6]
```diff
@@ -340,48 +339,40 @@ def parse_data(
...
+            IOType.FILEPATH: (lib.lyd_parse_data_path, get_in_data_str),
+        }
+
+        ret = parse_map[in_type][0](
+            self.cdata,
+            parse_map[in_type][1](),
+            fmt,
+            parser_flgs,
+            validation_flgs,
+            dnode,
```
Comment (rjarry) [inline libyang/context.py:364]: This is really ugly. Please avoid this kind of code at all costs :)

**Comment by jeanseb6wind** [comment]
abandonned, useless

**Pull request closed without merging**